src/Marten.Testing/Linq/query_with_inheritance.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Marten.Schema;
+using Marten.Services;
+using Shouldly;
+using Xunit;
+
+namespace Marten.Testing.Linq
+{
+    public interface ISmurf
+    {
+        string Ability { get; set; }
+        Guid Id { get; set; }
+    }
+    public class Smurf : ISmurf
+    {
+        public string Ability { get; set; }
+        public Guid Id { get; set; }
+    }
+    public interface IPapaSmurf : ISmurf{}
+    public class PapaSmurf : Smurf, IPapaSmurf{}
+    public class PapySmurf : Smurf, IPapaSmurf{}
+    public class BrainySmurf : PapaSmurf{}
+
+    public class query_with_inheritance_and_aliases : DocumentSessionFixture<NulloIdentityMap>
+    {
+        public query_with_inheritance_and_aliases()
+        {
+            StoreOptions(_ =>
+            {
+                _.Schema.For<ISmurf>()
+                    .AddSubclassHierarchy(
```

Comment:
I like this syntax. Someday we might wanna add some way, or just examples, to do this with type scanning or some other sort of dynamic Type discovery.